ripley/externals/redis.py

```diff
@@ -22,17 +22,44 @@
 "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
 ENHANCEMENTS, OR MODIFICATIONS.
 """
+import json
 
 from fakeredis import FakeStrictRedis
 from flask import current_app as app
 import redis
+from ripley import skip_when_pytest
+from ripley.api.errors import InternalServerError
 from rq import Connection, Queue, Worker
 from rq.job import Job, Retry
 
 
 redis_conn = None
 
 
+@skip_when_pytest()
+def cache_dict_object(cache_key, dict_object, expire_seconds=None):
+    if type(dict_object) is dict:
+        get_redis_conn(app)
+        redis_conn.set(cache_key, json.dumps(dict_object))
+        if expire_seconds:
+            redis_conn.expire(cache_key, time=expire_seconds)
+    else:
+        raise InternalServerError(f'Invalid object type: {type(dict_object)}')
+
+
+@skip_when_pytest()
+def fetch_cached_dict_object(cache_key):
+    get_redis_conn(app)
+    cached_dict_object = redis_conn.get(cache_key)
+    return None if cached_dict_object is None else json.loads(cached_dict_object)
+
+
+@skip_when_pytest()
+def remove_fetch_cached_dict_object(cache_key):
+    get_redis_conn(app)
+    redis_conn.delete(cache_key)
```

Comment:
`fakeredis` does not have clear documentation on supporting the JSON cache features of Redis. I'll revisit. In the meantime, we use `@skip_when_pytest`.